### DISCUSSION TITLE
Is a fresh start of container required after adding / modifying *.cf files?

### DISCUSSION THREAD

**Discussion by navyjosh** [docker-mailserver/docker-mailserver#3623] [Q&A] [answered]
Do I need to `docker-compose down && docker-compose up -d` after adding a new file to `./docker-data/dms/config`

For example, I add a `spamassassin-rules.cf` file to `./docker-data/dms/config`, do I need to down & up after?
What about for changes to a file that already exists?

**Comment by polarathene** [comment, marked as answer]
Usually changes will require this yes. Most of the configuration we support is handled at startup of the container only. Especially with content that sits in `/tmp/docker-mailserver` on the container, if the data isn't used by a service directly from there (it usually shouldn't), then it won't know about it until we copy it over to the proper place and adjust any necessary configuration to enable it (_which is what the startup scripts do_).

> What about for changes to a file that already exists?

We do have support for some of these configs.

It's not documented though, often called "change detection service" / "changedetector" / `check-for-changes.sh` (actual script name). Here's an example of explaining the SSL / TLS support: https://github.com/orgs/docker-mailserver/discussions/3416#discussioncomment-6349363

It is presently not active when using LDAP however. Other supported configs are mostly related to `setup ...` commands for mail accounts and aliases. So that when changes are made from that `setup` CLI you don't need to restart the container every time. Recently for upcoming v13 support was added for rspamd and DKIM generation.

**Reply by navyjosh** [reply]
Thank you, I was aware that setup script changes happen immediately, and that is what made me wonder about the cf files. This answers my question.
